ixml.ixml

```diff
@@ -8,8 +8,10 @@
       comment: -"{", (cchar; comment)*, -"}".
        -cchar: ~["{}"].
 
-         rule: (mark, S)?, name, S, ["=:"], S, -alts, ".".
+         rule: (mOrP, S)?, name, S, ["=:"], S, -alts, ".".
+        -mOrP: mark; pragma, mark?;
         @mark: ["@^-"].
+       pragma: "#", name, S, string.
```

Comment:
I wonder if parentheses, braces, or brackets would work better as delimiters than quotation marks.

For short pragmas, quotes are fine.  To make *num* as a `green` nonterminal and *var* as a `blue` nonterminal, I think it would look fine to write

    #my:color 'green' num: digit+, fractional-part?.
    #my:color 'blue' var: letter+.

But when the pragma contains an alternate formulation of a rule, it might be easier to read with distinctive start- and end-delimiters like () or [] or {}.  Cases like that also make me wonder again about representing a pragma in XML as an attribute.  

I am beginning to think that I need to look at or work out multiple examples first, before looking at the surface syntax.  For that purpose, this will do to start with.

Reply:
> I wonder if parentheses, braces, or brackets would work better as delimiters than quotation marks.

> Cases like that also make me wonder again about representing a pragma in XML as an attribute.

I think that the latter sentence cuts to the point of it.  If we represent as attributes, I think that it makes sense to re-use the string definition already in the grammar, and not have to worry about escaping characters, etc.  On the other hand, if that by its nature doesn't make sense, it's a signpost that attributes are the wrong way to implement.

> I am beginning to think that I need to look at or work out multiple examples first, before looking at the surface syntax. For that purpose, this will do to start with.

This sounds like the correct approach to me.